File: 0451-sort-characters-by-frequency/0451-sort-characters-by-frequency.java
class Solution {
    public String frequencySort(String s) {
        HashMap<Character,Integer> hm = new HashMap<>(); // hashMap with freq
        for(int i =0; i<s.length();i++) hm.put(s.charAt(i), hm.getOrDefault(s.charAt(i),0)+1);
        StringBuilder ans = new StringBuilder();
        List<Map.Entry<Character, Integer>> list = new ArrayList<>(hm.entrySet());
        list.sort(Map.Entry.comparingByValue());
        list.forEach( (point) -> { 
            for(int i =point.getValue()-1; i>=0;i--) ans.append(point.getKey());
         } );
        return ans.reverse().toString();
	}
}

File: 0451-sort-characters-by-frequency/0451-sort-characters-by-frequency.cpp
class Solution {
public:
    string frequencySort(string s) {
        map<char, int> hm;
        for(char ch : s) hm[ch]++;
        vector<pair<char,int>> vec;
        for(auto it : hm){
            vec.push_back(pair(it.first, it.second));
        }    
        // sort vector
        sort(vec.begin(), vec.end(), [](auto& a, auto& b){
            return a.second > b.second;
        });
        string str;
        for(auto it : vec){
            str.append(it.second, it.first);
        }
        return str;
    }
};